go/vt/vtgate/plan_execute.go

```diff
@@ -92,6 +92,7 @@ func (e *Executor) newExecute(
 		plan               *engine.Plan
 	)
 
+	var timeoutOnce sync.Once
```

Comment:
See explanation for why this is needed in a comment a few lines ahead.
```suggestion
	var timeoutOnce sync.Once
	var cancel context.CancelFunc
	defer func() {
	  if cancel != nil {
	    cancel()
	  }
	}()
```